upgrades/v3.0.5/README.md

```diff
@@ -0,0 +1,251 @@
+
+## Upgrade steps
+
+There are 2 major ways to upgrade a node:
+
+- Manual upgrade
+- Upgrade using [Cosmovisor](https://pkg.go.dev/cosmossdk.io/tools/cosmovisor)
+  - Either by manually preparing the new binary
+  - Or by using the auto-download functionality (this is not yet recommended)
+
+If you prefer to use Cosmovisor to upgrade, some preparation work is needed before upgrade.
+
+### Method I: Manual Upgrade
+
+Make sure Neutron v3.0.5 is installed by either downloading a [compatible binary](https://github.com/neutron-org/neutron/releases/tag/v3.0.5), or building from source. Building from source requires **Golang 1.21.x**.
+
+Run Neutron v3.0.2 till upgrade height, the node will panic:
+
+```shell
+ERR UPGRADE "v3.0.5" NEEDED at height:  10280000: upgrade to v3.0.5 and applying upgrade "v3.0.5" at height: 10280000
+```
+
+Stop the node, and switch the binary to **Neutron v3.0.5** and re-start by `neutrond start`.
+
+It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their node upgrades. After that, the chain can continue to produce blocks.
+
+### Method II: Upgrade using Cosmovisor
+
+### Manually preparing the binary
+
+##### Preparation
+
+Install the latest version of Cosmovisor (`1.5.0`):
+
+```shell
+go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@v1.5.0
+```
+
+**Verify Cosmovisor Version**
+
+```shell
+cosmovisor version
+cosmovisor version: v1.5.0
+```
+
+Create a cosmovisor folder:
+
+create a Cosmovisor folder inside `$NEUTRON_HOME` and move Neutron v3.0.5 into `$NEUTRON_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $NEUTRON_HOME/cosmovisor/genesis/bin
+cp $(which neutrond) $NEUTRON_HOME/cosmovisor/genesis/bin
+```
+
+build **Neutron v3.0.2**, and move neutrond v3.0.5 to `$NEUTRON_HOME/cosmovisor/upgrades/v3.0.5/bin`
+
+```shell
+mkdir -p  $NEUTRON_HOME/cosmovisor/upgrades/v3.0.5/bin
+cp $(which neutrond) $NEUTRON_HOME/cosmovisor/upgrades/v3.0.5/bin
+```
+
+Then you should get the following structure:
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── neutrond  #v3.0.2
+└── upgrades
+    └── v3.0.5
+        └── bin
+            └── neutrond  #v3.0.5
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=neutrond
+# please change to your own neutron home dir
+# please note `DAEMON_HOME` has to be absolute path
+export DAEMON_HOME=$NEUTRON_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node:
+
+```shell
+cosmovisor run  start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
+```
+
+Skipping the invariant checks is strongly encouraged since it decreases the upgrade time significantly and since there are some other improvements coming to the crisis module in the next release of the Cosmos SDK.
+
+#### Expected upgrade result
+
+When the upgrade block height is reached, Neutron will panic and stop:
+
+This may take a few minutes to a few hours.
```

Comment:
We can skip the sentence here since we have `upgrade duration` section, by the way the sectiones give us different outcome